django/db/models/functions/text.py

```diff
@@ -235,6 +238,49 @@ class RTrim(Transform):
     lookup_name = 'rtrim'
 
 
+class SHA1(MD5):
+    function = 'SHA1'
+    lookup_name = 'sha1'
+
+    def as_postgresql(self, compiler, connection, **extra_content):
+        # Note that the digest() function requires the pgcrypto extension.
+        return super().as_sql(
+            compiler,
+            connection,
+            template="ENCODE(DIGEST(%(expressions)s, '%(function)s'), 'hex')",
+            function=self.function.lower(),
+            **extra_content,
+        )
+
+
+class SHA224(SHA1):
+    function = 'SHA224'
+    lookup_name = 'sha224'
+
+    def as_mysql(self, compiler, connection, **extra_content):
+        return super().as_sql(
+            compiler,
+            connection,
+            template='SHA2(%%(expressions)s, %d)' % int(self.function[3:]),
```

Comment:
I think we can remove casting to `int`, i.e.
```python
template='SHA2(%%(expressions)s, %s)' % self.function[3:],
```

Reply:
Agreed. If wrong, it'll blow up anyway!